cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt

```diff
@@ -837,39 +840,70 @@ class PythonFrontendTest : BaseTest() {
         val forStmt = p.statements[1] as? ForEachStatement
         assertNotNull(forStmt)
 
-        val forVariable = forStmt.variable as? InitializerListExpression
+        val forVariable = forStmt.variable as? Reference
         assertNotNull(forVariable)
-        assertEquals(3, forVariable.initializers.size)
-        val t1Decl = forVariable.initializers[0] as? Reference
-        val t2Decl = forVariable.initializers[1] as? Reference
-        val t3Decl = forVariable.initializers[2] as? Reference
-        assertNotNull(t1Decl)
-        assertNotNull(t2Decl)
-        assertNotNull(t3Decl)
-        // TODO no refersTo
+        val forVarDecl =
+            p.declarations.firstOrNull {
+                it.name.localName.contains((PythonHandler.LOOP_VAR_PREFIX))
+            }
+        assertNotNull(forVarDecl)
+        assertEquals(forVarDecl, forVariable.refersTo)
 
         val iter = forStmt.iterable as? Reference
         assertNotNull(iter)
         assertEquals(testDeclaration, iter.refersTo)
 
         val forBody = forStmt.statement as? Block
         assertNotNull(forBody)
-        assertEquals(1, forBody.statements.size)
+        assertEquals(2, forBody.statements.size) // loop var assign and print stmt
+
+        /*
+        We model the 3 loop variables
+
+        ```
+        for t1, t2, t3 in ...
+        ```
+
+        implicitly as follows:
+
+        ```
+        for tempVar in ...:
+          t1, t2, t3 = tempVar
+          rest of the loop
+        ```
+         */
+        val forVariableImplicitStmt = forBody.statements.first() as? AssignExpression
+        assertNotNull(forVariableImplicitStmt)
+        assertEquals("=", forVariableImplicitStmt.operatorCode)
+        assertEquals(forStmt.variable, forVariableImplicitStmt.rhs.first())
```

Comment:
try to avoid first(), use firstOrNull(), otherwise the test will crash with an NPE rather than triggering the assertion.